src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/UniqueAccessCodeServiceImpl.java

```diff
@@ -195,7 +159,7 @@ public UniqueAccessCodeDTO linkUACCase(String uacHash, UACLinkRequestDTO request
 
     // if the uac indicates that the UAC is for a HI, we need to link the UAC to a new HI case
     // instead of the HH case
-    if (uac.getCaseType().equals(CaseType.HH.name())
+    if (primaryCase.getCaseType().equals(CaseType.HH.name())
```

Comment:
I have to confess to not understanding why we now look at the case type of the primaryCase instead of previously using the uac. However the comment still talks about the uac.